src/Microsoft.OpenApi/Extensions/OpenApiElementExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. 
+
+using System;
+using System.Collections.Generic;
+using Microsoft.OpenApi.Interfaces;
+using Microsoft.OpenApi.Models;
+using Microsoft.OpenApi.Services;
+using Microsoft.OpenApi.Validations;
+
+namespace Microsoft.OpenApi.Extensions
+{
+    /// <summary>
+    /// Extension methods that apply across all OpenAPIElements
+    /// </summary>
+    public static class OpenApiElementExtensions
+    {
+        /// <summary>
+        /// Validate element and all child elements
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="element"></param>
+        /// <returns></returns>
+        public static IEnumerable<ValidationError> Validate(this IOpenApiElement element)         {
```

Comment:
So, for the customer/developer, he will check:

1. The return is not null
2. The return has elements.

```C#
OpenApiDocument doc = new OpenApiDocument();
var errors = doc.Validate();

if (errors == null || !errors.Any())
{
    // validation passed.
}
else
{
   // validation failed.
}
```

Reply:
Checking errors.Any() should be sufficient because returning IEnumerable<T> as null is just evil.  I will add documentation to convey the fact that the function will never return null.